Update bandersnatch to latest packaging (22.0) to unblock tox 4.0

CI has shown (with tox needing new versions) that we will need some refactors to move to the new 22.0 versions of packaging.

PR: https://github.com/pypa/bandersnatch/pull/1309/files

Example error:
```
==================================== ERRORS ====================================
____ ERROR collecting src/bandersnatch/tests/plugins/test_latest_release.py ____
ImportError while importing test module '/home/runner/work/bandersnatch/bandersnatch/src/bandersnatch/tests/plugins/test_latest_release.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
src/bandersnatch/tests/plugins/test_latest_release.py:11: in <module>
    from bandersnatch_filter_plugins import latest_name
src/bandersnatch_filter_plugins/latest_name.py:5: in <module>
    from packaging.version import LegacyVersion, Version, parse
E   ImportError: cannot import name 'LegacyVersion' from 'packaging.version' (/home/runner/work/bandersnatch/bandersnatch/.tox/py3/lib/python3.8/site-packages/packaging/version.py)
=========================== short test summary info ============================
ERROR src/bandersnatch/tests/plugins/test_latest_release.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 2.06s ===============================
py3: exit 2 (2.68 seconds) /home/runner/work/bandersnatch/bandersnatch> coverage run -m pytest --strict-markers pid=2252
.pkg: _exit> python /opt/hostedtoolcache/Python/3.8.15/x64/lib/python3.8/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  py3: FAIL code 2 (44.61=setup[41.94]+cmd[2.68] seconds)
  evaluation failed :( (44.82 seconds)
```

Lets get this passing before we move to >= tox 4.0

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -5,6 +5,10 @@
 - Format with black `--preview` style `PR #1313`
   - I contribute to black and want to help find bugs ...
 
+## Dropped Support
+
+- Dropped LegacyVersion support for latest_name plugin `PR #1315`
+
 # 6.0.1
 
 ## Bug Fixes

--- a/requirements.txt
+++ b/requirements.txt
@@ -9,7 +9,7 @@ humanfriendly==10.0
 idna==3.4
 lxml==4.9.1
 multidict==6.0.3
-packaging==21.3
+packaging==22.0
 pyparsing==3.0.9
 setuptools==65.6.3
 six==1.16.0

--- a/src/bandersnatch_filter_plugins/latest_name.py
+++ b/src/bandersnatch_filter_plugins/latest_name.py
@@ -1,6 +1,6 @@
 import logging
 from operator import itemgetter
-from typing import Dict, Iterator, Tuple, Union
+from typing import Dict, Iterator, Tuple
 
 from packaging.version import Version, parse
 

--- a/src/bandersnatch_filter_plugins/latest_name.py
+++ b/src/bandersnatch_filter_plugins/latest_name.py
@@ -2,7 +2,7 @@ import logging
 from operator import itemgetter
 from typing import Dict, Iterator, Tuple, Union
 
-from packaging.version import LegacyVersion, Version, parse
+from packaging.version import Version, parse
 
 from bandersnatch.filter import FilterReleasePlugin
 
@@ -45,7 +45,7 @@ class LatestReleaseFilter(FilterReleasePlugin):
         if self.keep == 0 or self.keep > len(releases):
             return True
 
-        versions_pair: Iterator[Tuple[Union[LegacyVersion, Version], str]] = map(
+        versions_pair: Iterator[Tuple[Version, str]] = map(
             lambda v: (parse(v), v), releases.keys()
         )
         # Sort all versions

